### ISSUE TITLE
General Fault in Windows for Workgroups 3.11 for Help Files

### ISSUE THREAD

**Issue by nickkeane** [open-watcom/open-watcom-v2#1266]
I have a Windows for Workgroups 3.11 instance running in 86Box for 16-bit development. I've installed the latest version of Open Watcom on there. Whenever I go to open a help (.hlp) file for Open Watcom or go to Help->Contents in the IDE, I get a general protection fault error:

https://pasteboard.co/EJpTZHBY2DxT.png

https://pasteboard.co/J0c3cpxpOh8J.png

By contrast, Microsoft Visual Basic 3.0 HLP files are working fine. The Open Watcom IDE also seems to be working otherwise.

Edit: Nevermind, creating a source file caused the IDE to crash.

**Comment by jmalak** [comment]
Thank you for your problem report.
It looks like 16-bit Windows help files are damaged.
I don't understand you note
"Edit: Nevermind, creating a source file caused the IDE to crash."
Please could you explain.

**Comment by nickkeane** [comment]
Yeah, the IDE looked like it was working, but it crashes when i create a new source file. When I create a project, set target environment to Win16 and image type as Executable, it creates the project. Then, when clicking Sources -> New Source and entering "main.c" and clicking Add or pressing the Enter key, the IDE crashes.

https://pasteboard.co/hog7EMkxgSPB.png

**Comment by jmalak** [comment]
Thanks for explanation.
Please create next issue for this problem.

**Comment by nickkeane** [comment]
The issue with the help files causing a General Protection Fault in WINHELP is still occuring.
Using the 1/3/2025 build of OpenWatcom 2.0 C for Windows 16-bit, when opening "Getting Started" or any OpenWatcom 2.0 C/C++ Tools Help file.

![image](https://github.com/user-attachments/assets/1ffa4890-249f-4b3a-ad3f-a54be830fd79)

![image](https://github.com/user-attachments/assets/6f138630-088c-4ebd-a49c-83bce3250a38)

Other Help files seem to load fine, like Visual Basic f.e.:

![image](https://github.com/user-attachments/assets/b2c1e81d-ac4f-4475-80f7-e829cb6dae55)

**Comment by Baron-von-Riedesel** [comment]
> It looks like 16-bit Windows help files are damaged.

They are shown without problems in Windows 98 SE. So it seems it's rather kind of a WinHelp version problem.

**Comment by nickkeane** [comment]
Yeah the help files installed w/ the Windows 98 SE 32-bit version and the 64-bit versions on Windows 11 seem to have working help files. Not sure why the help files installed with the 16-bit Win 3.1 version don't work. Do you know if they've ever worked?

**Comment by Baron-von-Riedesel** [comment]
> Not sure why the help files installed with the 16-bit Win 3.1 version don't work. 

AFAIU, OW has its very own help compiler, it cannot use the compiler supplied with MSVC v1.5 (HC31.EXE). First thing to check, IMO, is if the help files supplied with OW 1.9 also crash.

**Comment by nickkeane** [comment]
It looks like those *.hlp files are working:

<img width="812" alt="Image" src="https://github.com/user-attachments/assets/33033df8-26c1-4cde-9945-f533b10c3110" />

Help contents are greyed out in IDE though, so I have to open them from the File Manager:

<img width="339" alt="Image" src="https://github.com/user-attachments/assets/8992a155-c527-48c4-8c51-c4bdfd9fbcbc" />

Using the [open-watcom-c-dos-1.9.exe](https://github.com/open-watcom/open-watcom-1.9/releases/download/ow1.9/open-watcom-c-dos-1.9.exe) installer w/ Win16 as Host, DOS and Win16 as Targets.

**Comment by jmalak** [comment]
It is definitely bug in OW help compiler.
Older OW 2.0 .hlp files work OK on 16-bit Windows.
I will fix it.

**Comment by jmalak** [comment]
Please download new on-line help files for Windows 3.x (*.hlp in binw subdirectory) after OW rebuild and check.
It looks like 16-bit Windows help viewer has some limit on Copyright text that Help Compiler set maximal length of text to 55 bytes.

**Comment by Baron-von-Riedesel** [comment]
While you are at it, there are problems with the Win16 version:

- Readme.txt and all .BAT, .CFG and .DBG files in BINW have just LF ending, not CR/LF ( and all files in H and samples as well ).
- Link for Readme.txt is broken ( there's no space between "NOTEPAD" and "C:\WATCOM\README.TXT"

**Comment by nickkeane** [comment]
I've re-installed Open Watcom 2.x with the new installer. The help files are working now.

**Comment by nickkeane** [comment]
I just noticed the additional issue w/ the Read Me program link.
The working directory of it is similarly wrong to the program path as well, and should probably be blank, and causing an error:

![Image](https://github.com/user-attachments/assets/25298fcc-a179-49cf-9b32-4ea03825fe5f)

**Comment by jmalak** [comment]
To CRLF problem, OW use by default LF only for all text files and all OW tools works with LF or CRLF end-of-line.
It will require to fix only batch files by installer.
In the past it was used for CRLF systems only that it was not a problem, now with regards to mixed installation OS script files will need fix LF/CRLF.
Anyway it is issue if you copy installation between systems with different EOL or it must be defined by target OS (run-time) and file type that it is not host OS dependent.

**Comment by Baron-von-Riedesel** [comment]
> It will require to fix only batch files by installer.

Ok, but at least Readme.txt should also be fixed - because Notepad.exe also has problems with LF line endings.

**Comment by jmalak** [comment]
OK we will fix Readme.txt EOL to CRLF for Windows (16-bit).
Please, if you know about other files please add it to this issue.

**Comment by jmalak** [comment]
the change in installer for file type "b" means to convert file EOL to CRLF for DOS and Windows.
The type "b" is now used for readme file.
